src/compiler/checker.ts

```diff
@@ -12073,7 +12073,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function addInheritedMembers(symbols: SymbolTable, baseSymbols: Symbol[]) {
         for (const s of baseSymbols) {
-            if (!symbols.has(s.escapedName) && !isStaticPrivateIdentifierProperty(s)) {
+            if (isStaticPrivateIdentifierProperty(s)) {
+                continue;
+            }
+            const derived = symbols.get(s.escapedName);
+            if (derived) {
+                // non-constructor/static-block assignment declarations are ignored here; they're not treated as overrides
+                if (derived.valueDeclaration

```

Comment:
Do you only need to do these checks if the valueDeclaration is in a JS file?

Reply:
Oh wait, only JS files have symbols whose valueDeclaration is a BinaryExpression. If TS had such a symbol, we'd want it to follow these rules too.
In fact, isConstructorDeclaredProperty checks all the right things so I don't have to do it here.